### ISSUE TITLE
Error in basis set

### ISSUE THREAD

**Issue by jslefche** [jslefche/piecewiseSEM#51]
Looks like `topSort` from `ggm` may be generating a weird adjacency matrix, but the basis set is definitely wrong. Need to look into this!

```
library(piecewiseSEM)


data = data.frame(
  rep = runif(100),
  acc = runif(100),
  rep = runif(100),
  cap = runif(100),
  comm = runif(100),
  infl = runif(100),
  know = runif(100),
  out = runif(100)
)


modelList = list(
  lm(rep ~ acc, data),
  lm(out ~ know + cap + comm + infl + rep + acc, data),
  lm(cap ~ know + comm, data)
)

sem.fit(modelList, data)
```

**Comment by jslefche** [comment]
Yeah, this is definitely from `topSort` since I get the same output from `ggm`:

```
psem.bs = sem.basis.set(modelList)

library(ggm)

ggm.bs = basiSet(DAG(rep ~ acc, out ~ know + cap + comm + infl + rep + acc, cap ~ know + comm))

sapply(1:length(psem.bs), function(i) rbind(psem.bs[[i]], ggm.bs[[i]])) #all the same
```

**Comment by jslefche** [comment]
Ok re-wrote `topSort` as `sort.dag` . MAYBE this fixed this (???)

**Comment by jslefche** [comment]
Yeah, this works *better* but still not best:

```
data = data.frame(
  a = runif(100),
  b = runif(100),
  c = runif(100),
  d = runif(100),
  e = runif(100),
  f = runif(100)
)

modelList = list(
  lm(b ~ a, data),
  lm(d ~ b + c, data),
  lm(e ~ c + d, data),
  lm(f ~ e, data)
)

sem.basis.set(modelList)

```

[[1]]
[1] "a" "d" "b" "c" XXX NO GOOD XXX should be "d" "b" "a"

[[2]]
[1] "a" "e" "c" "d" GOOD

[[3]]
[1] "a" "f" "e" GOOD

[[4]]
[1] "c" "b" "a" XXX NO GOOD XXX should be "a" "c" "b"

[[5]]
[1] "c" "f" "e" XXX NO GOOD XXX shuld be "c" "f" "b" "d" "e"

[[6]]
[1] "b" "e" "a" "c" "d" GOOD

[[7]]
[1] "b" "f" "a" "e" GOOD

[[8]]
[1] "d" "f" "c" "b" "e" XXX NO GOOD XXX should be "d" "f" "e"

**Comment by jslefche** [comment]
Ok the example above was wrong but anyways, think I got this again. Going to leave open because I probably didn't

**Comment by jslefche** [comment]
Fixed this in 2.0
